spec/system/employee_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Employee account', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  let!(:employee) { build(:employee) }
+
+  it 'sign up and sign in' do
+    visit root_path
+    click_link 'Sign Up'
+    fill_in 'Email', with: employee.email
+    fill_in 'Password', with: employee.password
+    fill_in 'Password confirmation', with: employee.password
+    click_button 'Sign up'
+    expect(page).to have_text('Welcome! You have signed up successfully.')
+    click_link 'Log Out'
+
+    # let!(:employee) { FactoryBot.attributes_for(:employee) }
+
+    # it 'sign up and sign in' do
+    # visit root_path
+    # click_link 'Sign Up'
+    # fill_in 'Email', with: employee
+    # fill_in 'Password', with: employee
+    # fill_in 'Password confirmation', with: employee
+    # click_button 'Sign up'
+    # expect(page).to have_text('Welcome! You have signed up successfully.')
+    # click_link 'Log Out'
```

Comment:
`build` creates an object, but does not save it in db, and since it is and object, you can access it's attributes with employee.email
'attributes_for' creates hash of attributes, so if you want to access one of attributes, you have to use employee[:email]

if you prefer working with `build` instead of `attributes_for` it's perfectly fine

As regards commented code - I think the below should be working:

```suggestion
      let(:employee_attributes) { attributes_for(:employee) }

     it 'sign up and sign in' do
     visit root_path
     click_link 'Sign Up'
     fill_in 'Email', with: employee[:email]
     fill_in 'Password', with: employee[:password]
     fill_in 'Password confirmation', with: employee[:password]
     click_button 'Sign up'
     expect(page).to have_text('Welcome! You have signed up successfully.')
     click_link 'Log Out'
```

Reply:
Unfortunately it throws an error
**"undefined local variable or method` employee 'for # <RSpec :: ExampleGroups :: EmployeeAccount: 0x0000000120f47f60> "**

I will try to figure it out but for now I will stay on `build` :)

Reply:
of course, becasue I ahve made a mistake :) there should be `employee_attributes`, not `employee` in lines 27,28 and 29

Reply:
I changed the code but it's the same error 
' let(:employee_attributes) { attributes_for(:employee) }
  it 'sign up and sign in' do
    visit root_path
    click_link 'Sign Up'
    fill_in 'Email', with: employee_attributes[:email]
    fill_in 'Password', with: employee_attributes[:password]
    fill_in 'Password confirmation', with: employee_attributes[:password]'

Reply:
ok, I will test it out tomorrow then, and get back with solution :)

Reply:
I checked it, and below code is working fine
My guess is you might have renamed variable for the login part - below commented section 

```
# frozen_string_literal: true

require 'rails_helper'

RSpec.describe 'Employee account', type: :system do
  before do
    driven_by(:rack_test)
  end

  let!(:employee_attributes) { attributes_for(:employee) }

  it 'sign up and sign in' do
    visit root_path
    click_link 'Sign Up'
    fill_in 'Email', with: employee_attributes[:email]
    fill_in 'Password', with: employee_attributes[:password]
    fill_in 'Password confirmation', with: employee_attributes[:password]
    click_button 'Sign up'
    expect(page).to have_text('Welcome! You have signed up successfully.')
    click_link 'Log Out'

    click_link 'Log In'
    fill_in 'Email', with: employee_attributes[:email]
    fill_in 'Password', with: employee_attributes[:password]
    click_button 'Log in'
    expect(page).to have_content 'Signed in successfully.'
  end
end

```

Reply:
You're right about the "log in" section, I didn't change the variable for employee, it was late ;).

In employee_spec, I used attributes_for, and in kudo_spec I left 'create' and 'build'.

It was interesting and I learned a lot from it, thank you :)

Reply:
tbh. I personally just use build, but attributes_for is also a way to go. My only issue with the attributes_For is that it skips the model's associations, but it might not be an issue now 👍